clang/runtime/dpct-rt/include/lapack_utils.hpp.inc

```diff
@@ -0,0 +1,141 @@
+// DPCT_COMMENT // DPCT_CODE
+// DPCT_COMMENT some code
+// DPCT_COMMENT // DPCT_LABEL_END
+// DPCT_COMMENT
+// DPCT_COMMENT Example2:
+// DPCT_COMMENT // DPCT_LABEL_BEGIN|FeatureNameDef|[Namespace]
+// DPCT_COMMENT // DPCT_DEPENDENCY_BEGIN
+// DPCT_COMMENT // FileID|FeatureNameRef
+// DPCT_COMMENT [// FileID|FeatureNameRef]
+// DPCT_COMMENT ...
+// DPCT_COMMENT // DPCT_DEPENDENCY_END
+// DPCT_COMMENT // DPCT_CODE
+// DPCT_COMMENT some code
+// DPCT_COMMENT // DPCT_LABEL_END
+// DPCT_COMMENT
+// DPCT_COMMENT For header file including dependency, please use predefined feature name:
+// DPCT_COMMENT   local_include_dependency: dpct helper files
+// DPCT_COMMENT   non_local_include_dependency: other header files
+
+// DPCT_LABEL_BEGIN|local_include_dependency|
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+// DPCT_LABEL_END
+#include "memory.hpp"
+#include "util.hpp"
+
+// DPCT_LABEL_BEGIN|non_local_include_dependency|
+// DPCT_DEPENDENCY_EMPTY
+// DPCT_CODE
+#include <oneapi/mkl.hpp>
+#include <sycl/sycl.hpp>
+// DPCT_LABEL_END
+
+namespace dpct {
+namespace lapack {
+// DPCT_LABEL_BEGIN|sygvd|dpct::lapack
+// DPCT_DEPENDENCY_BEGIN
+// Memory|get_buffer_T|UsmNone
+// DPCT_DEPENDENCY_END
+// DPCT_CODE
+/// Computes all eigenvalues and, optionally, eigenvectors of a real generalized
+/// symmetric definite eigenproblem using a divide and conquer method.
+/// \return Returns 0 if no synchronous exception, otherwise returns 1.
+/// \param [in] queue Device queue where calculations will be performed.
+/// \param [in] itype Must be 1 or 2 or 3. Specifies the problem type to be solved.
+/// \param [in] jobz Must be job::novec or job::vec.
+/// \param [in] uplo Must be uplo::upper or uplo::lower.
+/// \param [in] n The order of the matrices A and B.
+/// \param [in,out] a The symmetric matrix A.
+/// \param [in] lda The leading dimension of matrix A.
+/// \param [in,out] b The symmetric matrix B.
+/// \param [in] ldb The leading dimension of matrix B.
+/// \param [out] w Eigenvalues.
+/// \param [in] scratchpad Scratchpad memory to be used by the routine
+/// for storing intermediate results.
+/// \param [in] scratchpad_size Size of scratchpad memory as a number of
+/// floating point elements of type T.
+/// \param [out] info If lapack synchronous exception is caught, the value
+/// returned from info() method of the exception is set to \p info. If other
+/// SYCL synchronous exception is caught, -1 is set to \p info.
+template <typename T>
+inline int sygvd(sycl::queue &queue, std::int64_t itype, oneapi::mkl::job jobz,
+                 oneapi::mkl::uplo uplo, int n, T *a, int lda, T *b, int ldb,
+                 T *w, T *scratchpad, int scratchpad_size, int *info) {
+#ifdef DPCT_USM_LEVEL_NONE
+  auto info_buf = get_buffer<int>(info);
+  auto a_buffer = get_buffer<T>(a);
+  auto b_buffer = get_buffer<T>(b);
+  auto w_buffer = get_buffer<T>(w);
+  auto scratchpad_buffer = get_buffer<T>(scratchpad);
+  int info_val = 0;
```

Comment:
There seems to be no need in new variable, just use info. Here `*info` should be initialized with zero.